src/OpenSSL/SSL.py

```diff
@@ -1399,6 +1399,24 @@ def get_cert_store(self):
         pystore._store = store
         return pystore
 
+    def set_cert_store(self, store):
+        """
+        Set the certificate store for the context.
+        :param store: A X509Store object.
+        :return: None
+        """
+        try:
+            _lib.SSL_CTX_set_cert_store(self._context, store._store)
+            # The store is now owned by the context, so we need to
+            # remove the gc free in the object. We do this after the
+            # set since set may not exist.
+            _ffi.gc(store._store, None)
```

Comment:
This seems like the very wrong way to do this. We should be bumping the refcount here, not getting rid of the GC. Doing it this way means it's totally wrong if you put the same store on 2x ssl contexts

Reply:
We don't have that binding 😂 Good catch on the multiple adds though. We can't support this without adding the upref binding then.